packages/react-i18n/src/decorator.tsx

```diff
@@ -11,13 +12,14 @@ export interface WithI18nProps {
 export function withI18n(i18nOptions?: RegisterOptions) {
   return <OwnProps, C>(
     WrappedComponent: React.ComponentType<OwnProps & WithI18nProps> & C,
-  ): React.ComponentType<OwnProps> & C => {
+  ): React.ComponentType<OwnProps> &
+    NonReactStatics<typeof WrappedComponent> => {
     function WithTranslations(props: OwnProps) {
       const [i18n, ShareTranslations] = useI18n(i18nOptions);
 
       return (
         <ShareTranslations>
-          <WrappedComponent {...props} i18n={i18n} />
+          <WrappedComponent {...props as any} i18n={i18n} />
```

Comment:
What was the error here?

Reply:
It complains about this without casting to any. I don't have a proper solution for it yet.

```
(parameter) WrappedComponent: (React.ComponentClass<OwnProps & WithI18nProps, any> & C) | (React.FunctionComponent<OwnProps & WithI18nProps> & C)
Type 'OwnProps & { i18n: I18n; }' is not assignable to type '(IntrinsicAttributes & LibraryManagedAttributes<ComponentClass<OwnProps & WithI18nProps, any> & C, PropsWithChildren<OwnProps & WithI18nProps>>) | (IntrinsicAttributes & LibraryManagedAttributes<...>)'.
  Type 'OwnProps & { i18n: I18n; }' is not assignable to type 'IntrinsicAttributes & LibraryManagedAttributes<FunctionComponent<OwnProps & WithI18nProps> & C, PropsWithChildren<OwnProps & WithI18nProps>>'.
    Type 'OwnProps & { i18n: I18n; }' is not assignable to type 'LibraryManagedAttributes<FunctionComponent<OwnProps & WithI18nProps> & C, PropsWithChildren<OwnProps & WithI18nProps>>'.ts(2322)
```